### PR TITLE
Add github action to build on push

### PR DESCRIPTION
**Pull request by cohensh**
Enforce that the gradlew code builds for a PR

### CODE REVIEW

**Comment by ewpratten** [comment]
FYI, I arranged a special agreement with some GitHub staff back in 2018 that gives this specific organization some additional permissions that regular GitHub accounts don't have.

It appears that someone removed my permissions to view the billing settings, so I can't check the exact details.

My memory of the agreement is that we are allowed unlimited free organization members (usually costs $5/user/month), and we have some crazy high GitHub Actions cap that allows us to run much longer and more frequent builds that regular users can.

Basically, what I'm saying is, don't worry about sticking resource intensive operations in the CI pipeline. Our account should be able to handle it.

**Review by ewpratten — APPROVED** [review]

**Comment by awpratten** [comment]
Moved ownership to Tiet where it should have always been.  Students come and go.  They should not be owners of this repo
